runtime/datastructures/src/decls.rs

```diff
@@ -56,6 +58,36 @@ impl PartialEq for KElem {
   }
 }
 
+impl PartialOrd for KElem {
+    fn partial_cmp(&self, other: &KElem) -> Option<Ordering> {
+        let ret: i64 = unsafe {
+            hook_KORD_cmp(*self.0.get(), *other.0.get())
+        };
+        if ret < 0 {
+            Some(Ordering::Less)
+        } else if ret == 0 {
+            Some(Ordering::Equal)
+        } else {
+            Some(Ordering::Greater)
+        }
+    }
+}
+
+impl Ord for KElem {
+    fn cmp(&self, other: &KElem) -> Ordering {
```

Comment:
Do we actually need this function? The Eq trait is implemented by means of implicitly calling PartialEq and doesn't actually need to duplicate code. I'm willing to bet you can do the same thing with Ord and just delete this method

Reply:
I tried removing it, it doesn't work. I think the issue is that PartialOrd returns an Option, while Ord does not. It's kind of annoying.  What I can do, however, is call cmp with unwrap to reduce code duplication.

Reply:
Yea, this make more sense. The reason Eq requires no implementation is the only difference with PartialEQ is it's possible that there exists b such that b != b. By implementing Eq you are guaranteeing that isn't true.

With ParitalOrd you are dealing with different types, and partial orders are just much more common than PartialEQ, which is relegated primarily to NaN in IEEE floating point.